File: test-ui/resources/usuario-service.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */


//var app = angular.module('mainApp', []);

app.service('UsuarioService', function($http){
            this.square = function(a) {
               return a * a ; 
            }
            
            var listaUsuarios = {
              async2: function() {
                var promise = $http.get('http://localhost:5000/api/user').then(function (response) {
                  console.log(response);
                  return response.data;
                });
                return promise;
              }
            };
            return listaUsuarios;    

            
    
});

File: test-service/src/main/java/com/test/model/api/services/UsuarioService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.test.model.api.services;

import com.test.model.api.model.Usuario;
import java.util.List;

/**
 *
 * @author lprates
 */
public interface UsuarioService {
    
	Usuario findById(Long id);

	Usuario findByNome(String name);

	void saveUsuario(Usuario user);

	void updateUsuario(Usuario user);

	void deleteUsuarioById(Long id);

	void deleteAllUsuarios();

	List<Usuario> findAllUsuarios();

	boolean isUsuarioExist(Usuario user);
    
}
